### PR TITLE
Fixed Observable subscribe to accept both object and function signatures

### PR DESCRIPTION
**Pull request by sdesai**
@jhusain @michaelbpaulson - For review.

After recent changes to remove Rx from the get flow, discovered that the
http datasource's Observable.subscribe wasn't implemented to accept both the
Observer object interface and separate onNext, onError and onCompleted.

With Rx, the observer was always the normalized Observer object interface.
Without Rx, it was separate onNext, onError, onCompleted callbacks.

Fixes https://github.com/Netflix/falcor/issues/512 and passes the integration
test in https://github.com/Netflix/falcor/pull/515

### CODE REVIEW

File: src/request.js [hunk 1]
```diff
@@ -91,7 +105,7 @@ function request(method, options, context) {
           //
           // The json response type can be ignored if not supported, because JSON payloads are
           // parsed on the client-side regardless.
-          if (responseType !== 'json') {
+          if (config.responseType !== 'json') {
```
Comment (sdesai) [inline src/request.js]: Unrelated change. My editor lint picked up the undefined reference.

File: src/request.js [hunk 2]
```diff
@@ -3,11 +3,23 @@ var getXMLHttpRequest = require('./getXMLHttpRequest');
...
 
 Observable.create = function(subscribe) {
   var o = new Observable();
+
   o.subscribe = function(observer) {
+
+    if (typeof observer === 'function') {
+        observer = {
+            onNext: observer,
+            onError: (arguments[1] || noop),
```
Comment (jhusain) [inline src/request.js]: Use named argument so that we don't need to use the expensive arguments object.
